near-sdk/src/json_types/account.rs

```diff
@@ -0,0 +1,92 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::Serialize;
+use std::convert::{TryFrom, TryInto};
+
+use crate::env::is_valid_account_id;
+use crate::AccountId;
+
+/// Helper class to validate account ID during serialization and deserializiation
+#[derive(
+    Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize, Serialize,
+)]
+pub struct ValidAccountId(AccountId);
+
+impl ValidAccountId {
+    fn is_valid(&self) -> bool {
+        is_valid_account_id(&self.0.as_bytes())
+    }
+
+    pub fn account_id(&self) -> &AccountId {
+        &self.0
+    }
+
+    pub fn into_account_id(self) -> AccountId {
```

Comment:
Seems like `impl From<ValidAccountId> for AccountId` already implements this functionality.

Reply:
Ok. Should we rename `account_id()` into `.as_ref()`?